pkg/mvtxdaemon/server/service/rater/rater.go

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rater
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/common/expfmt"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	sharedqueue "github.com/numaproj/numaflow/pkg/shared/queue"
+)
+
+const CountWindow = time.Second * 10
+const MonoVtxReadMetricName = "monovtx_read_total"
+
+// MonoVtxRatable is the interface for the Rater struct.
+type MonoVtxRatable interface {
```

Comment:
Just a question out of curiosity, how is this vertex rater different from the pipeline rater? Thanks

Reply:
In the pipeline rater we calculate the rate per partition per pod.
In this scenario, we do not have such a concept, hence each pod just has a single metric value.